### PR TITLE
Add Sample federated authenticator

### PR DESCRIPTION
**Pull request by GANGANI**
## Purpose
Add Sample federated authenticator
Doc PR: https://github.com/wso2/docs-is/pull/2527

### CODE REVIEW

File: authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (pamodaaw) [inline authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml:3]: wrong year

File: authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticatorConstants.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public static final String CLIENT_SECRET = "ClientSecret";
+    public static final String CALLBACK_URL = "callbackUrl";
+    public static final String OAUTH2_AUTHZ_URL = "OAuth2AuthzEPUrl";
+    public static final String OAUTH2_TOKEN_URL = "OAuth2TokenEPUrl";
+
+    public static final String HTTP_ORIGIN_HEADER = "Origin";
+    public static final String SUB = "sub";
+
+    public static final String OIDC_DIALECT = "http://wso2.org/oidc/claim";
+}
```
Comment (pamodaaw) [inline authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticatorConstants.java:42]: missing new line

Comment (GANGANI) [reply authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticatorConstants.java]: Fixed

File: authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml (hunk 1 above)
Comment (GANGANI) [reply authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml]: Fixed

File: authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
+# Writing a Custom Federated Authenticator
+
+A custom federated authenticator can be written to authenticate a user with an external system. 
+So the external system can be any Identity provider such as Facebook, Twitter, Google and Yahoo. In 
+this guide another Identity Server has been used as a federated authenticator, which is called as partner 
+identity server throughout this guide. It is possible to use the extension points available in the WSO2 
+Identity Server to create custom federated authenticators.
```
Comment (mefarazath) [inline authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/README.md:7]: A diagram explaining the flow would be nice to have

Comment (GANGANI) [reply authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/README.md:7]: Added a diagram

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**